File: python/P215.py
class Solution(object):
    def findKthLargest(self, nums, k):
        ''' direct method
        return sorted(nums)[::-1][k-1]        
        '''
        heap = []
        for num in nums:
            heapq.heappush(heap, num)
        for i in range(len(nums)-k+1):
            element = heapq.heappop(heap)
        return element

File: java/P215.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>(k+1);
        for (int num:nums) {
            pq.offer(num);
            if (pq.size() == k+1) {
                pq.poll();
            }
        }
        return pq.poll();
    }
}